src/Features/Core/Portable/StackTraceExplorer/StackFrameParserHelpers.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    internal class StackFrameParserHelpers
+    {
+        /// <summary>
+        /// Makes sure that the string at least somewhat resembles the correct form.
+        /// Does not check validity on class or method identifiers
+        /// Example line:
+        /// at ConsoleApp4.MyClass.ThrowAtOne(p1, p2,) 
+        ///   |-------------------||--------||-------| 
+        ///           Class          Method    Args   
+        /// </summary>
+        /// <remarks>
+        /// See https://docs.microsoft.com/en-us/dotnet/api/system.environment.stacktrace for more information
+        /// on expected stacktrace form
+        /// </remarks>
+
+        public static bool TryParseMethodSignature(string line, int start, int end, out TextSpan classSpan, out TextSpan methodSpan, out TextSpan argsSpan)
```

Comment:
note: using the term 'class' is not really appropriate as these might be structs/interfaces as well.  use `fullTypeNameSpan` (and try to state that consistently everywhere).  For 'method' i think it's ok as i would expect these to always be methods of some form.

Reply:
Renamed 'class' to 'type'